sphinx/source/start_network.rst

```diff
@@ -44,37 +44,63 @@ To join an existing network, other nodes should be started with the ``join`` opt
+The joining node takes the certificate of the existing network to join via ``--network-cert-file`` and initiates an enclave-to-enclave TLS connection to an existing node of the network as specified by ``--target-rpc-address``. Once the join protocol [#remote_attestation]_ completes, the joining node becomes part of the network as a backup (see :ref:`Ledger replication` for more details on consensus protocols).
 
 .. note:: When starting up the network or when joining an existing network, the network secrets required to decrypt the ledger are sealed and written to a file so that the network can later be recovered. See :ref:`Catastrophic Recovery` for more details on how to recover a crashed network.
 
-Opening a network
-~~~~~~~~~~~~~~~~~
+Opening a network to users
+--------------------------
 
-Once a CCF network is successfully started and an acceptable number of nodes have joined, members should vote to open the network to users. Users will then be able to execute transactions on the business logic defined by the enclave file (``--enclave-file`` option to ``cchost``).
+Adding users
+~~~~~~~~~~~~
 
-For example, the first member may decide to make a proposal to open the network:
+Once a CCF network is successfully started and an acceptable number of nodes have joined, members should vote to open the network to users. First, the certificates of trusted users should be registered in CCF via the member governance interface. For example, the first member may decide to make a proposal to add a new user (here, ``user_cert`` is the PEM certificate of the user -- see :ref:`Cryptography` for a list of supported algorithms):
 
 .. code-block:: bash
 
-    $ memberclient --cert member1_cert --privk member1_privk --rpc-address rpc_ip:rpc_port --ca network_cert open_network
+    $ memberclient --cert member1_cert --privk member1_privk --rpc-address rpc_ip:rpc_port --ca network_cert add_user --user_cert user_cert
     {"commit":4,"global_commit":3,"id":0,"jsonrpc":"2.0","result":{"completed":false,"id":0},"term":2}
 
 Other members are then allowed to vote for the proposal, using the proposal ID returned to the proposer member (here ``0``, as per ``"result":{"completed":false,"id":0}``).
 
 .. code-block:: bash
 
-    memberclient --cert member2_cert --privk member2_privk --rpc-address rpc_ip:rpc_port --ca network_cert vote --proposal-id 0 --accept
-    {"commit":6,"global_commit":3,"id":0,"jsonrpc":"2.0","result":true,"term":2}
+    $ memberclient --cert member2_cert --privk member2_privk --rpc-address rpc_ip:rpc_port --ca network_cert vote --proposal-id 0 --accept
```

Comment:
Do we want to support the option of multiple users being added by the member at once, instead of adding them one by one?

Reply:
I think this should be supported by adding some logic around CCF but not in CCF itself. And if members of CCF wants this feature, we can add it then.

Reply:
This is supported on the server side, and it's one of the things we'll improve once we've switched to a lighter HTTP + JSON templates client. The *client binaries are effectively maintenance only.

Reply:
I was mostly thinking of it in the proposal context, of a member wanting to add a bunch of users at once, and the other members can review and accept that proposal with all the users at once, instead of voting for each one separately. But I guess then if you want to reject one of the users but not all of them you need to reject the whole proposal. But in practise it might be useful. Just a thought.